extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/RestrictedGuacamoleTunnelService.java

```diff
@@ -180,8 +188,15 @@ protected ModeledConnection acquire(RemoteAuthenticatedUser user,
             @Override
             public int compare(ModeledConnection a, ModeledConnection b) {
 
-                return getActiveConnections(a).size()
-                     - getActiveConnections(b).size();
+                // Get connection weight for the two systems being compared.                
+                int weightA = a.getConnectionWeight().intValue();
+                int weightB = b.getConnectionWeight().intValue();
+
+                // Get current active connections, add 1 to both to avoid calculations with 0.
```

Comment:
Why? What do you have against 0? ;)

Especially considering that you:

> ...  removed the checking for > 0. Since anything < 1 gets eliminated from consideration later on, it doesn't really matter how those connections sort, so there isn't much point in checking.